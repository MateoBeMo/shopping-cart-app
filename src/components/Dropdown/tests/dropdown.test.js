import React from 'react';
/**Components */
import DropDown from '../Dropdown';
/**Testing librery */
import { fireEvent, cleanup } from '@testing-library/react';
import { axe } from 'jest-axe';
/**Setup */
import { renderWithReactIntl } from 'setupTests';

const onClick = jest.fn();
const locale = 'es';

const options = [
    {
        value: 'es',
        name: 'app.components.Dropdown.Language.Spanish',
    },
    {
        value: 'en',
        name: 'app.components.Dropdown.Language.English',
    },
];

const selected = 'Español';

describe('<Dropdown>', () => {
    afterEach(() => {
        cleanup();
    });

    test('Snapshot renders => it should rendered with the provided props', () => {
        const { container } = renderWithReactIntl(<DropDown options={options} value={locale} onClick={onClick} />);

        expect(container).toMatchSnapshot();
    });

    test('It should render with the provided props', () => {
        const { getByTestId } = renderWithReactIntl(<DropDown value={locale} options={options} onClick={onClick} />);

        expect(getByTestId('dropdown__selected-test')).toHaveTextContent(selected);
    });

    test('It should render options when click div button', () => {
        const { queryByText, getByTestId } = renderWithReactIntl(
            <DropDown value={locale} options={options} onClick={onClick} />,
        );

        const divButton = getByTestId('dropdown__selected-test');

        expect(queryByText(/Inglés/i)).not.toBeInTheDocument();

        fireEvent.click(divButton);

        expect(queryByText(/Inglés/i)).toBeInTheDocument();
    });

    test('It should call onclick function received by props when click over li element', () => {
        const { queryByText, getByTestId } = renderWithReactIntl(
            <DropDown value={locale} options={options} onClick={onClick} />,
        );

        const divButton = getByTestId('dropdown__selected-test');

        fireEvent.click(divButton);

        const listItem = queryByText(/Inglés/i);

        fireEvent.click(listItem);

        expect(onClick).toHaveBeenCalled();

        expect(onClick).toHaveBeenCalledTimes(1);
    });

    test('Should rendered with no options and with a information message instead', () => {
        const { queryByText, getByTestId } = renderWithReactIntl(
            <DropDown value={'en'} options={null} onClick={onClick} />,
        );
        const divButton = getByTestId('dropdown__selected-test');
        fireEvent.click(divButton);
        expect(queryByText(/la lista está vacía/i)).toBeInTheDocument();
    });

    test("Should rendered with the default value passed by props even if it doesn't match", () => {
        const { getByTestId } = renderWithReactIntl(<DropDown value={'fr'} options={options} onClick={onClick} />);
        expect(getByTestId('dropdown__selected-test')).toHaveTextContent('');
    });

    test('should not have basic accessibility issues', async () => {
        const { container } = renderWithReactIntl(<DropDown value={locale} options={options} onClick={onClick} />);
        const results = await axe(container);
        expect(results).toHaveNoViolations();
    });
});
