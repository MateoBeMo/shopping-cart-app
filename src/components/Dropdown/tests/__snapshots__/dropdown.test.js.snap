// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown> Snapshot renders => it should rendered with the provided props 1`] = `
<div>
  <div
    class="dropdown__wrapper"
  >
    <div
      class="dropDown__button__wrapper"
    >
      <div
        class="dropdown__button"
        data-testid="dropdown__selected-test"
      >
        <img
          alt="es-flag"
          src="ES-Spain-Flag-icon.png"
        />
        <span>
          Espa√±ol
        </span>
      </div>
      <span
        class="dropdown__arrow"
        style="pointer-events: none;"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chevron-down fa-w-14 "
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
